climateconnect/climateconnect
# Feat: Organization Social Media Links (Old version can maybe delete)

## Description

Closes #1071

Backend: 
Added a SocialMediaChannel & SocialMediaLink model. A channel is a constant that points to a social media site. The link binds this channel to an organization.  
Updated serializers: organization.py and created new ones in social_media.py. Modified organization_views for creating/editing organizations.

Frontend:
Users can click a + icon on a chip that reads add social media. This opens a selectfield dialog from which they can pick from 5 different social medias. Depending on the type of social media, the user will need to either insert the full link or just the handle. A second textfield will open and will display the full URL for the user if only the handle is required. Once added, a chip dispaying the icon and name, with the option to delete it via a close button, will appear on the edit page. When the chip is hovered, the url will display. When viewing the organization page, the social media links will be displayed via their icons that when clicked, will open a new tab with their social media page opened.
The website of the organization has been moved from the center to the left under their avatar. 
Also fixed where selecting empty field, the page doesn't crash for types or social media options.

TODOS: 
Update texts for German page

## Test plan
1. Create an organization
2. Go through the steps and in the detailed input screen, chip should be visible. Clicking the + icon should open a dialog in which you can select a type of social media write a valid url for that specific social media.
3. Click create after entering everything
4. Click "edit organization"
5. You can now also delete or modifiy existing urls

## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)
